Manipulators/BaseManipulator.cs

```diff
@@ -87,24 +122,185 @@ void OnCameraPreRender(Camera camera)
 		public void AdjustScale(Vector3 cameraPosition, Matrix4x4 worldToCameraMatrix)
 		{
 			var originalCameraPosition = cameraPosition;
-			
+
 			// Adjust size of manipulator while accounting for any non-standard cameras (e.g. scaling applied to the camera)
 			var manipulatorPosition = worldToCameraMatrix.MultiplyPoint3x4(transform.position);
 			cameraPosition = worldToCameraMatrix.MultiplyPoint3x4(cameraPosition);
 			var delta = worldToCameraMatrix.inverse.MultiplyPoint3x4(cameraPosition - manipulatorPosition) - originalCameraPosition;
 			transform.localScale = Vector3.one * delta.magnitude * k_BaseManipulatorSize;
 		}
 
-		protected void OnDragStarted()
+		protected virtual void UpdateHandleTip(BaseHandle handle, HandleEventData eventData, bool active)
+		{
+			var rayOrigin = eventData.rayOrigin;
+			HandleTip handleTip;
```

Comment:
there are a lot of handletips here, some that are handletips, and some that are named handletip but are renderers